kgl-math/src/commonMain/kotlin/com/kgl/math/Vectors.kt

```diff
@@ -0,0 +1,289 @@
+package com.kgl.math
+
+/**
+ * A generic two-component vector. Methods in this interface support only read-only access to the vector;
+ * read/write access is supported through the [MutableVector2] interface.
+ * @param T the type of component contained in the vector. The vector is invariant on its component type.
+ */
+interface Vector2<T> {
+
+	val x: T
+	val y: T
+
+	operator fun component1(): T = x
+	operator fun component2(): T = y
+
+	operator fun get(index: Int): T = when (index) {
+		0 -> x; 1 -> y; else -> throw IndexOutOfBoundsException()
+	}
+
+	fun copy(): Vector2<T>
+
+
+	// geometric
+
+	val length: T
+
+	fun normalize(): Vector2<T>
```

Comment:
I think there should be two variants of this function.
- `fun normalized(): Vector2<T>` - returns this vector but normalized.
- `fun normalize(): Unit` - normalizes this vector. (only for mutable variant)

Reply:
I was thinking that too but was unsure what to call it 👍

Reply:
implemented in new commit